build.gradle

```diff
@@ -1,181 +1,252 @@
+// src: https://github.com/gradle/gradle/issues/10708#issuecomment-566279825
+task rootJacocoMergedReport(type: JacocoReport) {
+  //dependsOn = subprojects.test
+  additionalSourceDirs.setFrom files(subprojects.sourceSets.main.allSource.srcDirs)
+  sourceDirectories.setFrom files(subprojects.sourceSets.main.allSource.srcDirs)
+  classDirectories.setFrom files(subprojects.sourceSets.main.output)
+  executionData.setFrom project.fileTree(dir: '.', include: '**/build/jacoco/test.exec')
+  reports {
+    xml.enabled true
+    html.enabled true
+  }
 }
-// Use the filter task as the input for compileJava
-compileJava.source = processSourceTokens.outputs
-
-// Font Width Info
-task copyMCFontExtractor(type: Copy) {
-    def path = project.configurations.compile.find {it.name.startsWith("mc-font-extractor") }
-    println path
-    from file(path)
-    // into file("src/main/resources")
-    // - Please keep this comment for future reference.
-    // - This is how we would do this if we weren't also adding build info (see processSourceTokens)
-    destinationDir file("src/main/resources/generated/")
-    rename 'mc-font-extractor-main-mojangles_width_data.json', 'mojangles_width_data.json'
+
+// Create a libs/modules folder with submodule jars
+task copySubprojectJars(type: Copy, dependsOn: subprojects.jar) {
+    // Copy subproject jar and sources
+    from(subprojects.jar) 
+    from(subprojects.sources)
+    into rootProject.file("build/libs/modules")
 }
 
+// Root build: run copySubprojectJars after build
+tasks.build.finalizedBy copySubprojectJars
+
+// Root build: run rootSources during build
+tasks.build.dependsOn rootSources
+
+// Root build: generate project wide test, coverage and javadoc results
+tasks.test.finalizedBy rootTestReport, rootJacocoMergedReport, javadoc
+
+// Root build: also run aggregateJavadoc after javadoc to generate project-wide javadocs
+tasks.javadoc.finalizedBy aggregateJavadoc
+
+// Javadoc fixes (pt. 2, see above entry in subprojects)
+aggregateJavadoc.options.encoding = "UTF-8"
+
+/*
+    Javadoc Fixes, part 3
+
+    Since JDK13 the javadoc html validator was changed, making it impossible to have javadocs
+    that are valid on both pre-13 and post-13 java versions.
+    The original workaround was suggested here: https://bugs.openjdk.java.net/browse/JDK-8223552
+    However, this workaround does not work for us since we are using aggregate-javadoc.
+    So, we are using a workaround for the workaround from:
+        original: https://blog.joda.org/2014/02/turning-off-doclint-in-jdk-8-javadoc.html
+        archived: https://web.archive.org/web/20210117193942/https://blog.joda.org/2014/02/turning-off-doclint-in-jdk-8-javadoc.html
+    which, in turn, references this: https://github.com/GPars/GPars/blob/7cddf7cf2fec1fd66ef800edccfc03315d078a2b/build.gradle#L209
+ */
+if (JavaVersion.current().isJava8Compatible()) {
+    allprojects {
+        tasks.withType(Javadoc) {
```

Comment:
Same as above, does `javadoc {}` not exist?

Reply:
See above comment